frontend/src/components/common/AddAdminModal.tsx

```diff
@@ -0,0 +1,189 @@
+      fontSize="18px"
+      width="320px"
+      height="48px"
+      textAlign="center"
+      backgroundColor="grey.100"
+      borderColor="grey.100"
+    />
+  );
+};
+
+const AddAdminModal = (): React.ReactElement => {
+  const { onOpen, onClose, isOpen } = useDisclosure();
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [email, setEmail] = useState("");
+  const [confirmEmail, setConfirmEmail] = useState("");
+  const [hasJumpMathEmail, setHasJumpMathEmail] = useState<boolean | null>(
+    null,
+  );
+
+  const onModalClose = () => {
+    setHasJumpMathEmail(null);
+    onClose();
+  };
+
+  const onAddAdminClick = () => {
+    if (email !== confirmEmail) {
+      console.log("emails don't match");
+      return;
+    }
+    console.log("creating admin with data");
+    console.log(
+      `first name: ${firstName}\nlast name: ${lastName}\nemail: ${email}`,
+    );
+    onModalClose();
+  };
+
+  return (
+    <>
+      <Button
+        rightIcon={<PlusOutlineIcon />}
+        variant="primary"
+        my={2}
+        onClick={onOpen}
+      >
+        Add Admin
+      </Button>
+      <Modal isOpen={isOpen} onClose={onModalClose} size="6xl" isCentered>
+        <ModalOverlay />
+        <ModalContent p={2}>
+          <ModalHeader>
+            <Text textStyle="subtitle1" color="grey.400">
+              Add Admin
+            </Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <FormControl isRequired>
+              <AddAdminFormLabel text="Name of Admin" />
+              <HStack direction="row" mt={6}>
+                <AddAdminTextInput
+                  validationType="text"
+                  placeholder="First Name"
+                  handleChange={setFirstName}
+                />
+                <AddAdminTextInput
+                  validationType="text"
+                  placeholder="Last Name"
+                  handleChange={setLastName}
+                />
+              </HStack>
+            </FormControl>
+            <FormControl isRequired as="fieldset" mt={8}>
+              <AddAdminFormLabel text="Does the user already have a Jump Math email address?" />
+              <RadioGroup
+                onChange={(val) => setHasJumpMathEmail(val === "yes")}
+                mt={5}
+              >
+                <HStack spacing="24px">
+                  <Radio value="no" mt={2} outlineColor="grey.300" size="lg">
+                    No
+                  </Radio>
+                  <Radio value="yes" outlineColor="grey.300" size="lg">
+                    Yes
+                  </Radio>
+                </HStack>
+              </RadioGroup>
+            </FormControl>
+            {hasJumpMathEmail !== null && (
+              <>
+                <FormControl isRequired mt={6}>
+                  <AddAdminFormLabel
+                    text={`Please enter their ${
```

Comment:
In the case that the user selects that they have a Jump Math email address can we validate that they use that email address? Also, if we could pre-fill the field with the Jump Math email address format (i.e `@jumpmath.org`)